### ISSUE TITLE
Guard against division by zero

### ISSUE THREAD

**Issue by kevinleewy** [UKPLab/sentence-transformers#625]
https://github.com/UKPLab/sentence-transformers/blob/f782b9c0c37a5921a2488c0c725bbf474fc7b207/sentence_transformers/util.py#L13

The sentence embedding of unrecognized words (using `average_word_embeddings_glove.840B.300d` model) will return a vector of zeros. If I then take the cosine distance using these zero embeddings I'll get a nan because the normalization step isn't guarded against division by zero. I recommend adding a small epsilon term in the denominator on lines 31 and 32.

**Comment by nreimers** [comment]
Thanks for pointing this out.

I updated the code and it will be part of the next release.

**Comment by kevinleewy** [comment]
Using `torch.nn.functional.normalize` is a great solution since it has a default epsilon value of `1e-12`! Great!
